### PR TITLE
fixed using history.pushState & window.location.search

### PR DESCRIPTION
**Pull request by lang-sky**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "browser": true// important so document & window are not undefined
   },
   "rules": {
+    "linebreak-style": 0,
```
Comment (ar-to) [inline .eslintrc]: probably add 
`"linebreak-style": 0,// linebreak-style: ["error", "windows"] is better but seems to also cause errors in linux

File: src/SortFilter/Core.js [hunk 2]
```diff
@@ -121,24 +122,35 @@ class Core extends Component {
...
    */
   sortByText = (text) => {
-    this.shuffle.filter(text);
+    // this.shuffle.filter(text);
+    window.location.search = text;
+    this.sortFromUrlSearch();
   }
 
+  sortFromUrlSearch = () => {
+    const searchParams = jsUcfirst(window.location.search.slice(1));
```
Comment (ar-to) [inline src/SortFilter/Core.js:146]: jsUcFirst is probably not necessary . Its usually not a good practice to use capital letters in the url. See [here](https://webmasters.stackexchange.com/questions/33255/google-and-querystring-parameter-capitalization)

File: src/SortFilter/Core.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Shuffle from 'shufflejs';
 import SortButtons from './SortButtons';
+import { jsUcfirst } from '../Utils/index';
```
Comment (ar-to) [inline src/SortFilter/Core.js]: see comment below

File: src/SortFilter/Core.js [hunk 4]
```diff
@@ -121,24 +122,35 @@ class Core extends Component {
...
   }
 
   /**
    * Sort by the specific text
    * @param {string} text
    */
   sortByText = (text) => {
-    this.shuffle.filter(text);
+    // this.shuffle.filter(text);
+    window.location.search = text;
```
Comment (ar-to) [inline src/SortFilter/Core.js]: shouldn't this be `history.pushState()`

**Review by ar-to — COMMENTED** [review]
**maybe** externalize the pushState into a Utils function

File: src/SortFilter/Core.js (hunk 2 above)
Comment (lang-sky) [reply src/SortFilter/Core.js:146]: It's necessary, cause the filter didn't work with 'city' or 'cat'.
It worked with 'City' or 'Cat'

**Comment by lang-sky** [comment]
will "Pull request again" with newer one.

**Comment by ar-to** [comment]
Fixes #13

**Pull request merged**